google/cloud/aiplatform/jobs.py

```diff
@@ -2495,6 +2501,9 @@ def update(
             model_deployment_monitoring_job=current_job,
             update_mask=field_mask_pb2.FieldMask(paths=update_mask),
         )
+        self._gca_resource = self.api_client.get_model_deployment_monitoring_job(
```

Comment:
The update call returns an `Operation`. The result of the `Operation` is the `ModelDeploymentMonitoringJob`. Reference:

https://github.com/googleapis/python-aiplatform/blob/6dc5f12c771c05efb4ec304054400b1b887a0980/google/cloud/aiplatform_v1/services/job_service/client.py#L3599

We should:
1. Block on this operation until we add optional_sync support (that can be captured as a TODO with reference to a ticket)
2. Assign the gca_resource to the result of that operation.

If we do not then the local object  likely becomes out of sync with the service after this update call. 

This should work:

```Python
operation = self.api_client.update_model_deployment_monitoring_job(
            model_deployment_monitoring_job=current_job,
            update_mask=field_mask_pb2.FieldMask(paths=update_mask),
        )

self._gca_resource = operation.result()
```

Reply:
done